backend/src/database/users.go

```diff
@@ -542,23 +543,93 @@ func (db *DB) GetUserProgramInfo(args *models.QueryContext, userId int) ([]model
+			return nil, NewDBError(err, "program_class_enrollments")
+		}
+		if err := base.Limit(args.PerPage).Offset(args.CalcOffset()).Scan(&userEnrollments).Error; err != nil {
+			return nil, NewDBError(err, "program_class_enrollments apply pagination")
+		}
+	} else {
+		base.Order("pc.name")
+		if err := base.Find(&userEnrollments).Error; err != nil {
+			return nil, NewDBError(err, "program_class_enrollments")
+		}
+
+	}
+	return userEnrollments, nil
+}
 
-	if err := base.Count(&args.Total).Error; err != nil {
+func (db *DB) GetUserProgramClassHistory(args *models.QueryContext, userId int) ([]models.ResidentProgramClassHistory, error) {
+	var residentProgramClassHistory []models.ResidentProgramClassHistory
+	err := db.WithContext(args.Ctx).
+		Table("program_class_enrollments pce").
+		Select(`p.name as program_name, pc.name AS class_name,
+		COALESCE(
+		(
+			SELECT STRING_AGG(pt.credit_type::text, ',')
+			FROM program_credit_types pt
+			WHERE pt.program_id = pc.program_id
+		), 
+		''
+		) AS credit_types,
+		pce.enrollment_status AS status,
+		pce.updated_at AS date_status_changed`).
+		Joins("INNER JOIN program_classes pc ON pce.class_id = pc.id").
+		Joins("INNER JOIN programs p ON p.id = pc.program_id").
+		Where("pce.user_id = ?", userId).
+		Order("pce.updated_at DESC").
+		Group("p.name, pc.name, pc.program_id, pce.enrollment_status, pce.updated_at").
+		Find(&residentProgramClassHistory).Error
+
+	if err != nil {
 		return nil, NewDBError(err, "program_class_enrollments")
 	}
-	if err := base.Limit(args.PerPage).Offset(args.CalcOffset()).Scan(&userEnrollments).Error; err != nil {
-		return nil, NewDBError(err, "program_class_enrollments apply pagination")
+
+	return residentProgramClassHistory, nil
+}
+
+func (db *DB) GetUserWeeklySchedule(args *models.QueryContext, userId int) ([]models.ResidentProgramClassWeeklySchedule, error) {
+	var residentWeeklySchedule []models.ResidentProgramClassWeeklySchedule
+
+	err := db.WithContext(args.Ctx).
+		Table("program_class_enrollments pce").
+		Select(`pc.name AS class_name,
+				pc.start_dt AS start_date,
+				COALESCE(pc.end_dt::text, 'N/A') AS end_date,
+	COALESCE(
+	(
+		SELECT STRING_AGG(pt.credit_type::text, ',')
+		FROM program_credit_types pt
+		WHERE pt.program_id = pc.program_id
+	), 
+	''
+	) AS credit_types,
+	pce.enrollment_status AS status,
+	pce.updated_at AS date_status_changed`).
+		Joins("INNER JOIN program_classes pc ON pce.class_id = pc.id").
+		Joins("INNER JOIN programs p ON p.id = pc.program_id").
+		Where("pce.user_id = ? ", userId).
+		Where("pc.start_dt BETWEEN CURRENT_DATE AND CURRENT_DATE + INTERVAL '7 days'").
+		Order("pce.updated_at DESC").
+		Group("pc.program_id, pc.name, pc.start_dt, pc.end_dt, pce.enrollment_status, pce.updated_at").
+		Find(&residentWeeklySchedule).Debug().Error
+
+	if err != nil {
+		return nil, NewDBError(err, "program_class_enrollments")
 	}
 
-	return userEnrollments, nil
+	return residentWeeklySchedule, nil
 }
```

Comment:
I think I have two questions about this function. They might contradict each other, but here goes.

1. If I understand this correctly, it's looking for Classes with a START DATE between today and 7 days. The purpose of the weekly schedule is so that a Resident can answer the question, "What classes do I have this week? Or Next week, etc." A class can start and meet daily, weekly, monthy over an extended period of time. 

2. Previous point aside...this function isn't much different than `GetUserProgramClassHistory` and `GetUserProgramInfo`, right...with the addition of the start_dt filter. Even if we keep it, it feels like a lot of extra duplicate code for that one additional filter.